File: tests/001-return_1/test_return_1.nim
import pymod

proc cfloatReturn1*():  cfloat {.exportpy.} = 1.0
proc cdoubleReturn1*(): cdouble {.exportpy.} = 1.0

#proc cscharReturn1*():  cschar {.exportpy.} = 1
proc cshortReturn1*():  cshort {.exportpy.} = 1
proc cintReturn1*():    cint {.exportpy.} = 1
proc clongReturn1*():   clong {.exportpy.} = 1

proc cushortReturn1*(): cushort {.exportpy.} = 1
proc cuintReturn1*():   cuint {.exportpy.} = 1
proc culongReturn1*():  culong {.exportpy.} = 1

proc floatReturn1*():   float {.exportpy.} = 1.0
proc float32Return1*(): float32 {.exportpy.} = 1.0
proc float64Return1*(): float64 {.exportpy.} = 1.0

proc intReturn1*():     int {.exportpy.} = 1
#proc int8Return1*():    int8 {.exportpy.} = 1
proc int16Return1*():   int16 {.exportpy.} = 1
proc int32Return1*():   int32 {.exportpy.} = 1
proc int64Return1*():   int64 {.exportpy.} = 1

proc uintReturn1*():    uint {.exportpy.} = 1
proc uint8Return1*():   uint8 {.exportpy.} = 1
proc uint16Return1*():  uint16 {.exportpy.} = 1
proc uint32Return1*():  uint32 {.exportpy.} = 1
proc uint64Return1*():  uint64 {.exportpy.} = 1

proc byteReturn1*():    byte {.exportpy.} = 1

proc ccharReturn1*():   cchar {.exportpy.} = '1'
proc charReturn1*():    char {.exportpy.} = '1'
proc cucharReturn1*():  cuchar {.exportpy.} = '1'
proc stringReturn1*():  string {.exportpy.} = "one"


initPyModule("_pymod_test",
    cfloatReturn1, cdoubleReturn1,
    #cscharReturn1,
    cshortReturn1, cintReturn1, clongReturn1,
    cushortReturn1, cuintReturn1, culongReturn1,
    floatReturn1, float32Return1, float64Return1,
    #int8Return1,
    intReturn1, int16Return1, int32Return1, int64Return1,
    uintReturn1, uint8Return1, uint16Return1, uint32Return1, uint64Return1,
    byteReturn1,
    ccharReturn1, charReturn1, cucharReturn1, stringReturn1)

File: tests/001-return_1/test_return_1.py
def test_0_compile_pymod_test_mod(pmgen_py_compile):
    pmgen_py_compile(__name__)


def test_cfloatReturn1(pymod_test_mod):
    print(dir(pymod_test_mod))
    assert pymod_test_mod.cfloatReturn1() == 1.0

def test_cdoubleReturn1(pymod_test_mod):
    assert pymod_test_mod.cdoubleReturn1() == 1.0


def test_cshortReturn1(pymod_test_mod):
    assert pymod_test_mod.cshortReturn1() == 1

def test_cintReturn1(pymod_test_mod):
    assert pymod_test_mod.cintReturn1() == 1

def test_clongReturn1(pymod_test_mod):
    assert pymod_test_mod.clongReturn1() == 1


def test_cushortReturn1(pymod_test_mod):
    assert pymod_test_mod.cushortReturn1() == 1

def test_cuintReturn1(pymod_test_mod):
    assert pymod_test_mod.cuintReturn1() == 1

def test_culongReturn1(pymod_test_mod):
    assert pymod_test_mod.culongReturn1() == 1


def test_floatReturn1(pymod_test_mod):
    assert pymod_test_mod.floatReturn1() == 1.0

def test_float32Return1(pymod_test_mod):
    assert pymod_test_mod.float32Return1() == 1.0

def test_float64Return1(pymod_test_mod):
    assert pymod_test_mod.float64Return1() == 1.0


def test_intReturn1(pymod_test_mod):
    assert pymod_test_mod.intReturn1() == 1

#def test_int8Return1(pymod_test_mod):
#    assert pymod_test_mod.int8Return1() == 1

def test_int16Return1(pymod_test_mod):
    assert pymod_test_mod.int16Return1() == 1

def test_int32Return1(pymod_test_mod):
    assert pymod_test_mod.int32Return1() == 1

def test_int64Return1(pymod_test_mod):
    assert pymod_test_mod.int64Return1() == 1


def test_uintReturn1(pymod_test_mod):
    assert pymod_test_mod.uintReturn1() == 1

def test_uint8Return1(pymod_test_mod):
    assert pymod_test_mod.uint8Return1() == 1

def test_uint16Return1(pymod_test_mod):
    assert pymod_test_mod.uint16Return1() == 1

def test_uint32Return1(pymod_test_mod):
    assert pymod_test_mod.uint32Return1() == 1

def test_uint64Return1(pymod_test_mod):
    assert pymod_test_mod.uint64Return1() == 1


#def test_boolReturn1(pymod_test_mod):
#    assert pymod_test_mod.boolReturn1() == True

def test_byteReturn1(pymod_test_mod):
    assert pymod_test_mod.byteReturn1() == 1


#def test_unicodeRuneReturn1(pymod_test_mod, python_major_version):
#    if python_major_version == 2:
#        assert pymod_test_mod.unicodeRuneReturn1() == u"a"
#    else:  # Python 3 or above: bytes vs strings, yay!
#        assert pymod_test_mod.unicodeRuneReturn1() == "a"

def test_ccharReturn1(pymod_test_mod, python_major_version):
    if python_major_version == 2:
        assert pymod_test_mod.ccharReturn1() == "a"
    else:  # Python 3 or above: bytes vs strings, yay!
        assert pymod_test_mod.ccharReturn1() == b"a"

def test_charReturn1(pymod_test_mod, python_major_version):
    if python_major_version == 2:
        assert pymod_test_mod.charReturn1() == "a"
    else:  # Python 3 or above: bytes vs strings, yay!
        assert pymod_test_mod.charReturn1() == b"a"

def test_stringReturn1(pymod_test_mod):
    assert pymod_test_mod.stringReturn1() == "one"


def test_intReturn1NoParensInDecl(pymod_test_mod):
    assert pymod_test_mod.intReturn1NoParensInDecl() == 1

